tcp-keepalive/internal/server/server.go

```diff
@@ -26,18 +29,31 @@ func initLogger() {
 	log = slog.New(slog.NewJSONHandler(os.Stdout, nil))
 }
 
-func NewServer(cfg *Config) (*Server, error) {
+func NewServer(cfg *Config, mcfg *metrics.Config) (*Server, error) {
```

Comment:
I think we don't have to receive pointer type as arguments.
If we accept pointer types, we have to check whether they are nil or not.

Reply:
@terassyi 
> I think we don't have to receive pointer type as arguments.

Yes, we don't have to use pointer types but I think we should use them because they have an advantage in terms of memory footprint compared to value types.

> If we accept pointer types, we have to check whether they are nil or not.

I agree. I'll add nil checks.